### PR TITLE
Dynamic Dashboard: Separate top performers from store stat section

### PR DESCRIPTION
**Pull request by itsmeichigo**
Part of #12487 

## Description

This PR separates the existing `storeStatsAndTopPerformers` dashboard card type to separate types `peformance` and `topPerformers`. A new view for the Top Performers section on the Dashboard screen has also been created - due to the large number of changes, the implementation for the top performer list will be added in a separate PR.

Note: The same config is being used for the selected time range for the store performance and top performer cards. We will need to separate the configs in a separate PR.

## Testing instructions

- Enable the feature flag `dynamicDashboard` and build the app.
- Log in to a store and confirm that the Top Performers section should be displayed if it's enabled.
- Switch between different time ranges, the time range text should be updated accordingly.
- Tap on View all store analytics button, the analytics hub should be displayed.

## Screenshots

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/196">18.3</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Dashboard/TopPerformers/TopPerformersDashboardView.swift [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        self.onViewAllAnalytics = onViewAllAnalytics
+    }
+
+    var body: some View {
+        VStack(alignment: .leading) {
+            header
+                .padding(.horizontal, Layout.padding)
+                .redacted(reason: viewModel.syncingData ? [.placeholder] : [])
+                .shimmering(active: viewModel.syncingData)
+
```
Comment (hafizrahman) [inline WooCommerce/Classes/ViewRelated/Dashboard/TopPerformers/TopPerformersDashboardView.swift:26]: design np: I think this part needs a bit more vertical spacing. Figma vs Simulator comparison below:

Screenshot 2024-04-18 at 14 11 52

File: WooCommerce/Classes/ViewRelated/Dashboard/TopPerformers/TopPerformersDashboardView.swift [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            timeRangeBar
+                .padding(.horizontal, Layout.padding)
+                .redacted(reason: viewModel.syncingData ? [.placeholder] : [])
+                .shimmering(active: viewModel.syncingData)
+
+            Divider()
+
+            // TODO
+
+            Divider()
```
Comment (hafizrahman) [inline WooCommerce/Classes/ViewRelated/Dashboard/TopPerformers/TopPerformersDashboardView.swift:36]: Can be added in this PR or in the next one, just a reminder that this divider needs a bit of left spacing according to Figma:

Screenshot 2024-04-18 at 14 16 31

**Review by hafizrahman — APPROVED** [review]
Great work, code looks nice to me! I left some small design comments but otherwise should be good to go after you check them.

File: WooCommerce/Classes/ViewRelated/Dashboard/TopPerformers/TopPerformersDashboardView.swift (hunk 1 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Dashboard/TopPerformers/TopPerformersDashboardView.swift:26]: Good catch! Updated the spacing in 5357cbcf683c6d91eec46bfac09d85ce9ef73603.

File: WooCommerce/Classes/ViewRelated/Dashboard/TopPerformers/TopPerformersDashboardView.swift (hunk 2 above)
Comment (itsmeichigo) [reply WooCommerce/Classes/ViewRelated/Dashboard/TopPerformers/TopPerformersDashboardView.swift:36]: Also updated the padding in 5357cbcf683c6d91eec46bfac09d85ce9ef73603.

**Pull request merged**